backend/src/v4/models/chatEvent.js

```diff
@@ -217,34 +223,41 @@ const subscribeToV5Events = () => {
 			}
 		}
 		if(message) {
-			const Mailer = require("../mailer/mailer");
 			const notes = await notifications.insertModelUpdatedFailedNotifications(teamspace, model, user, message);
 			notes.forEach((note) => upsertedNotification(null, note));
 
 			if(!userErr) {
-				const attachments = [];
-				const path = require("path");
-				const sharedSpacePath = require("../config").cn_queue.shared_storage;
-				const sharedDir = path.join(sharedSpacePath, corId);
-				const files = require("fs").readdirSync(sharedDir);
-				files.forEach((file) => {
-					if(file.endsWith(".log")) {
-						attachments.push({
-							filename: file,
-							path: path.join(sharedDir, file)
+				try {
+					const archive = archiver("zip", { zlib: { level: 1 } });
+					const sharedDir = path.join(sharedSpacePath, corId);
+					const files = fs.readdirSync(sharedDir);
+
+					files.forEach((file) => {
+						if (file.endsWith(".log")) {
+							archive.file(path.join(sharedDir, file), { name: file });
+						}
+					});
+
+					const zipPath = path.join(sharedDir, "logs.zip");
+					const output = fs.createWriteStream(zipPath);
+
+					output.on("close", () => {
+						Mailer.sendImportError({
+							account: teamspace,
+							model,
+							username: user,
+							err: message,
+							corID: corId,
+							bouncerErr: errCode,
+							attachments: [{ filename: "logs.zip", path: zipPath }]
 						});
-					}
-				});
-
-				Mailer.sendImportError({
-					account: teamspace,
-					model,
-					username: user,
-					err: message,
-					corID: corId,
-					bouncerErr: errCode,
-					attachments
-				});
+					});
+
+					archive.pipe(output);
```

Comment:
I would create the output stream and pipe  to output before L235, so it can get the stream as soon as it is available.